libbeat/outputs/otelconsumer/otelconsumer.go

```diff
@@ -0,0 +1,159 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package otelconsumer
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/outputs"
+	"github.com/elastic/beats/v7/libbeat/publisher"
+	"github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+func init() {
+	outputs.RegisterType("otelconsumer", makeOtelConsumer)
+}
+
+type otelConsumer struct {
+	observer     outputs.Observer
+	logsConsumer consumer.Logs
+	beatInfo     beat.Info
+}
+
+func makeOtelConsumer(_ outputs.IndexManager, beat beat.Info, observer outputs.Observer, cfg *config.C) (outputs.Group, error) {
+
+	out := &otelConsumer{
+		observer:     observer,
+		logsConsumer: beat.LogConsumer,
+		beatInfo:     beat,
+	}
+
+	ocConfig := defaultConfig()
+	if err := cfg.Unpack(&ocConfig); err != nil {
+		return outputs.Fail(err)
+	}
+	return outputs.Success(ocConfig.Queue, -1, 0, nil, out)
+}
+
+// Close is a noop for otelconsumer
+func (out *otelConsumer) Close() error {
+	return nil
+}
+
+// Publish converts Beat events to Otel format and send to the next otel consumer
+func (out *otelConsumer) Publish(ctx context.Context, batch publisher.Batch) error {
+	switch {
+	case out.logsConsumer != nil:
+		return out.logsPublish(ctx, batch)
+	default:
+		panic(fmt.Errorf("an otel consumer must be specified"))
+	}
+}
+
+func (out *otelConsumer) logsPublish(_ context.Context, batch publisher.Batch) error {
```

Comment:
Is there a way to unit test this?

Same comment for `mapstrToPcommonMap` which definitely looks like it could have exhaustive tests.

Reply:
I see some tests in receiver_test.go but it would be nice to have something closer to the output itself considering how small it is.

Reply:
> Same comment for `mapstrToPcommonMap` which definitely looks like it could have exhaustive tests.

Added some tests for `mapstrToPcommonMap`, which I think we might want refactor in the future with go generics.